app/src/main/java/io/appium/uiautomator2/utils/XMLHierarchy.java

```diff
@@ -78,14 +65,14 @@ public static Node getFormattedXMLDoc() throws UiAutomator2Exception {
     public static Node formatXMLInput(InputSource input) {
         XPath xpath = XPathFactory.newInstance().newXPath();
 
-        Node root = null;
+        Node root;
         try {
             root = (Node) xpath.evaluate("/", input, XPathConstants.NODE);
         } catch (XPathExpressionException e) {
-            throw new RuntimeException("Could not read xml hierarchy: ", e);
+            throw new InvalidSelectorException("Could not read xml hierarchy: ", e);
```

Comment:
```java
throw new UiAutomator2Exception("Could not obtain root element in xml hierarchy:", e);
```

Reply:
XPathExpressionException is about incorrect xpath selector, isn't it?

Reply:
Yes. But it will be strange to get ```INVALID_SELECTOR``` response on ```/wd/hub/session/:sessionId/source``` endpoint. Furthermore we should never reach this catch block cuz ```/``` is always a valid xpath expression.